ckanext/canada/extract.py

```diff
@@ -0,0 +1,97 @@
+from yaml import load
+from six import string_types
+
+
+def _get_line_number(fileobj, phrase, encoding):
+    line = fileobj.readline().decode(encoding)
+    num = 1
+    while line:
+        if phrase in line:
+            return num
+        line = fileobj.readline().decode(encoding)
+        num += 1
+    return None
+
+
+def _extract_string(fileobj, key, value, encoding, comments):
+    fileobj.seek(0)
```

Comment:
instead of repeatedly re-parsing the file ourselves for every string can we use something like the answer to https://stackoverflow.com/questions/13319067/parsing-yaml-return-with-line-number ?

Reply:
Seems like all of these are based on yaml Nodes that can have keys to them. So far, it seems that the string constructor really is forcing the return of a single value...but will keep trying

Reply:
@wardi okay well it is a bit strange. Sadly it was really impossible to record the line numbers for str type nodes because of how they are outputted and encoded into the dict later on.

But I was able to override the scalar constructor which the str constructor builtin calls. And then I just recorded line numbers into a class variable. And then return the loader with the `load` method and access the class variable whilst yielding the strings to get the line numbers.